src/pepr/operator/crd/validators/package-validator.ts

```diff
@@ -141,5 +141,24 @@ export async function validator(req: PeprValidateRequest<UDSPackage>) {
     }
   }
 
+  const monitors = pkg.spec?.monitor ?? [];
+
+  // Ensure serviceMonitors use a unique description or selector/portName used for generating the resource name
+  const monitorNames = new Set<string>();
+
+  for (const monitor of monitors) {
```

Comment:
Didn't think about this before but I think there's another edge case where podmonitors and servicemonitors _should_ be allowed to have overlapping names, but this would deny them. We may need separate sets per type and check the `kind` in this loop.